### ISSUE TITLE
Package name is confusing

### ISSUE THREAD

**Issue by dreua** [claird/PyPDF4#30]
I was very confused that after installing PyPDF4 there was no package called PyPDF4 or pypdf4. At the very least this should be prominently advertised in the readme but acutally I would like you, the maintainers, to reconsider the decision to name the package pypdf instead of pypdf4.

The change was introduced in https://github.com/claird/PyPDF4/pull/14#issuecomment-420702997 (Commit: https://github.com/claird/PyPDF4/commit/7346ac1c35f6fa0b581d16acfa32f2f8e204f73b ) without further discussion. While I understand the advantage described in the commit message and comment, I would like you to consider the confusion that it causes for users. It could also lead to name conflicts e.g. when users want to compare differend versions of pypdf or use software that requires different versions. 

[PEP 423: Use a single Name](https://www.python.org/dev/peps/pep-0423/#use-a-single-name):
> Distribute only one package (or only one module) per project, and use package (or module) name as project name.

I'm not sure about this PEPs "deferred" status, but the arguments there are convincing to me.

If you want to stick with pypdf as package name anyway I would suggest to start versioning with major version 4 instead of 0 proposed in #16 to hint what project this package belongs to and avoid collisions.

**Comment by newnone** [comment]
TL;DR The organizational/repository name should (possibly) become PyPDF, the package/code name stay `pypdf` along major future versions of PyPDF.

Thank you for bringing this up. My own personal idea (IIRC) when submitting those name changes was to abstract away the version details from the distribution, i.e. have PyPDF look as possible as we can the same to the user either in version 1, 2, (3), 4 etc. Specifically, I saw no reason to split identity when shifting from v2 to v4 (although we obviously changed the package name from `PyPDF2` to the simpler `pypdf`) and, would it one day upgrade to v5, the package name ought to stay `pypdf` again.

Currently, PyPI lists different "brands" of PyPDF. That causes confusion to me:
![screenshot_2019-03-03 search results](https://user-images.githubusercontent.com/5630708/53693352-9a570780-3d9f-11e9-95ec-d7adae6d7ff9.png)

**I myself am surprised I haven't brought this up before**, but as it seems the custom for most projects, I advocate the use of a single, `PyPDF` name, for the project. Starting from now on, different version shall be distinguished by... its very versioning number. There shall ideally be one single strand of history in PyPDF (the project, organizational name), and speaking of PyPDF2, PyPDF4 etc. should only happen at implementation level. Obviously, as hinted to by [PEP 423](https://www.python.org/dev/peps/pep-0423/#how-to-rename-a-project) official, legacy repositories should be kept the way they are. The organizational change should only apply with the newer versions of PyPDF.

I have only been a free contributor and, seeing no other advancement from others, I have tried to make my own. Unfortunately feedback and replies from the core mantainer(s) have been so slow to motivate code changes even before I got explicit confirm and thorough discussions.

**Comment by newnone** [comment]
To sum it up, I see two primary choices:
* Stick PyPDF to the PyPDFX naming-scheme across major versions, and reflect this style to the package/code level (in that sense, the package name should become `pypdf4` as @dreua seems to suggest)
* Use PyPDF (w/o numbers attached) as the organizational/repository name for all the possible future versions of the project. The repository itself should at this point be renamed to `claird/PyPDF` to maintain consistency; the naming-scheme of the package (code-level) need not be changed

I'm sorry I haven't done all the possible research on the subject matter, but I advocate the use of the second choice. What would happen for PyPDF5? A new repository, a new entry on PyPI?

**Comment by xilopaint** [comment]
There are two confusing things to me:

1) this repo is a fork of PyPDF2 instead of an independent repo;

2) the first release tag of PyPDF4 is 1.27.0 instead of 1.0.0.

**Comment by newnone** [comment]
@xilopaint as for 2., however we choose the `1.27.0` value is inappropriate and should be changed.

While I have favored starting from version `0.1.0` in some past occasions, if the repo name is to be changed to just `PyPDF` then `4.0.0` should be the starting point.

With regard to 1. I can't say what was in Cameron's mind while forking the old PyPDF2 repository, other than he wished a clean start for the project.

**Comment by xilopaint** [comment]
> While I have favored starting from version `0.1.0` in some past occasions, if the repo name is to be changed to just `PyPDF` then `4.0.0` should be the starting point.

This would make sense. In any case 1.27.0 looks the worst possible choice.

**Comment by newnone** [comment]
>This would make sense. In any case 1.27.0 looks the worst possible choice.

AFAIK, it's the only choice admissible for a project that hasn't received proper time commitment from core maintainers :-/.

**Comment by xilopaint** [comment]
> AFAIK, it's the only choice admissible for a project that hasn't received proper time commitment from core maintainers :-/.

If you have good knowledge on the codebase, why don't start a new repository? This project started with the promise of daily updates on Twitter. The last one dates from six months ago.

PyPDF2 (as its predecessor PyPDF) is a great project. Many people still use it to learn Python and manipulate PDF files. PyPDF3 and PyPDF4 don't honour that legacy. It's time to move on.

**Comment by newnone** [comment]
>If you have good knowledge on the codebase, why don't start a new repository? This project started with the promise of daily updates on Twitter. The last one dates from six months ago.

Because it puts high responsibilities on one's shoulders. I mean, I can definitely do one such thing and get rewards out of the community's satisfaction and possible donations (something that the current PyPDF could definitely establish even now), but even then it would not be my top priority.

What I can do is, maybe, ensure a few contributions now and then.

**Comment by dreua** [comment]
Sorry for the late answer, I kind of forgot about it:
1. Yes, I do suggest that the package name should be pypdf4. (Zen of Python No. 2: "Explicit is better than implicit.") I don't care at what version it starts as long the package name is pypdf4.
2. To be honest I think it would be best if all effort could go into PyPdf2 since that is what most people and distributions use. (We should work with the maintainer of PyPdf2 and try to make this possible.) PyPdf2 in the hands of an active community or in the hands of [codeshelter](https://www.codeshelter.co/) would be a good path for the future. Creating more and more forks spreading out issue reports and contributor time is not.

**Comment by MasterOdin** [comment]
> To be honest I think it would be best if all effort could go into PyPdf2 since that is what most people and distributions use. (We should work with the maintainer of PyPdf2 and try to make this possible.) PyPdf2 in the hands of an active community or in the hands of codeshelter would be a good path for the future. Creating more and more forks spreading out issue reports and contributor time is not.

I agree it would be ideal to continue PyPDF2 as it has the most name recognition and just go with a BC-breaking v2 release that the PyPDF3 and PyPDF4 forks were promising to do, but it is very unclear where @claird and @mstamy2 are actually planning to do on maintenance since it seems like both make a yearly post and then disappear and would prefer both of these forks to just die already.

**Comment by hellerbarde** [comment]
@claird @mstamy2 is there any news on this? I am the maintainer of [stapler](https://github.com/hellerbarde/stapler) and I'd love to point my dependencies to an actively developed PyPDF* library, because I get bug reports sometimes that I can't fix on my side.

**Comment by dreua** [comment]
Since pdfarranger switched to pikepdf I am no longer interested in this issue. It is not pure python but well maintained and I have yet to find a PDF which it can't read.

**Comment by xilopaint** [comment]
> @claird @mstamy2 is there any news on this? I am the maintainer of [stapler](https://github.com/hellerbarde/stapler) and I'd love to point my dependencies to an actively developed PyPDF* library, because I get bug reports sometimes that I can't fix on my side.

There're no such thing as an actively maintained PyPDF library.

**Comment by hellerbarde** [comment]
@dreua I see

@xilopaint Yeah, I figured.
